x-pack/test/apm_api_integration/tests/traces/trace_by_id.spec.ts

```diff
@@ -4,84 +4,221 @@
-        )
-      ).toMatchInline(`
-        Array [
-          "/dashboard (transaction)",
-          "GET /api/stats (transaction)",
-          "APIRestController#topProducts (transaction)",
-          "Parsing the document, executing sync. scripts (span)",
-          "GET /api/products/top (span)",
-          "GET /api/stats (span)",
-          "Requesting and receiving the document (span)",
-          "SELECT FROM customers (span)",
-          "SELECT FROM order_lines (span)",
-          "http://opbeans-frontend:3000/static/css/main.7bd7c5e8.css (span)",
-          "SELECT FROM products (span)",
-          "SELECT FROM orders (span)",
-          "SELECT FROM order_lines (span)",
-          "Making a connection to the server (span)",
-          "Fire \\"load\\" event (span)",
-          "empty query (span)",
-        ]
-      `);
-      expectSnapshot(response.body.exceedsMax).toMatchInline(`false`);
+    after(() => synthtraceEsClient.clean());
+
+    describe('return trace', () => {
+      let traces: Awaited<ReturnType<typeof fetchTraces>>['body'];
+      before(async () => {
+        const response = await fetchTraces({
+          traceId: serviceATraceId,
+          query: { start: new Date(start).toISOString(), end: new Date(end).toISOString() },
+        });
+        expect(response.status).to.eql(200);
+        traces = response.body;
+      });
+      it('returns some errors', () => {
+        expect(traces.errorDocs.length).to.be.greaterThan(0);
+        expect(traces.errorDocs[0].error.exception?.[0].message).to.eql(
+          '[ResponseError] index_not_found_exception'
+        );
+      });
+
+      it('returns some trace docs', () => {
+        expect(traces.traceDocs.length).to.be.greaterThan(0);
+        expect(
+          traces.traceDocs.map((item) => {
+            if (item.span && 'name' in item.span) {
+              return item.span.name;
+            }
+            if (item.transaction && 'name' in item.transaction) {
+              return item.transaction.name;
+            }
+          })
+        ).to.eql(['GET /apple 🍏', 'get_green_apple_🍏']);
+      });
     });
   });
+
+  registry.when(
+    'Trace with span links',
+    { config: 'basic', archives: ['apm_mappings_only_8.0.0'] },
+    () => {
+      let serviceATraceId: string;
+      let serviceBTraceId: string;
+      before(async () => {
+        const instanceJava = apm.service('Service-A', 'production', 'java').instance('instance-a');
+        const serviceAEvents = timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) => {
+            return instanceJava
+              .transaction('GET /service_A')
+              .timestamp(timestamp)
+              .duration(1000)
+              .success()
+              .children(
+                instanceJava
+                  .span('get_service_A', 'db', 'elasticsearch')
+                  .defaults({ 'span.links': generateExternalSpanLinks(3) })
```

Comment:
I think it'd help tremendously if these are more realistic (and appropriately-named) scenarios. Messaging systems maybe?

Reply:
serviceA and serviceB might also be a good candidate for renaming.

Reply:
I created new tests here https://github.com/elastic/kibana/pull/126630/commits/651bf45f68ea0236a889afeb760d73586821374d#diff-5a56fd212eecdb6f4035475e1af2689b8e5f99ee97b8073f08964356ca0a1555R12